sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -515,20 +524,105 @@ export class ApplicationOfferingChangeRequestService {
    * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
    * @param applicationOfferingChangeRequestStatus the application offering change request status to be updated.
    * @param studentConsent student consent to approve the application offering change request.
+   * @param userId user id of the user making the request.
    */
   async updateApplicationOfferingChangeRequest(
     applicationOfferingChangeRequestId: number,
     applicationOfferingChangeRequestStatus: ApplicationOfferingChangeRequestStatus,
     studentConsent: boolean,
+    userId: number,
   ): Promise<void> {
+    const auditUser = { id: userId } as User;
+    const currentDate = new Date();
     await this.applicationOfferingChangeRequestRepo.update(
       {
         id: applicationOfferingChangeRequestId,
       },
       {
         applicationOfferingChangeRequestStatus,
         studentConsent,
+        studentActionDate: currentDate,
+        modifier: auditUser,
+        updatedAt: currentDate,
       },
     );
   }
+
+  /**
+   * Update the application offering change request status for the given application offering change request id.
+   * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
+   * @param applicationId application id related to the application offering change request.
+   * @param offeringId requested offering id related to this application offering change request.
+   * @param studentId student id of the student related to the application offering change request.
+   * @param applicationOfferingChangeRequestStatus the application offering change request status to be updated.
+   * @param assessmentNote note added while updating the application offering change request.
+   * @param userId user id of the user making the approve or decline request.
+   */
+  async assessApplicationOfferingChangeRequest(
+    applicationOfferingChangeRequestId: number,
+    applicationId: number,
+    offeringId: number,
+    studentId: number,
+    applicationOfferingChangeRequestStatus: ApplicationOfferingChangeRequestStatus,
```

Comment:
if the plan is not to send the payload as it is mentioned in https://github.com/bcgov/SIMS/pull/2392/files#r1353116289, then you can restrict the status type here

Reply:
Isn't it restricted by the STO as you mentioned [](https://github.com/bcgov/SIMS/pull/2392/files#r1353116289)

Reply:
here I was not mentioning the DTO validation, This service is only for ministry actions right? so, you can restrict the status here, so that, it will be clear that this service is only used for end status updates and devs won't be using this service by mistake in the future.